### ISSUE TITLE
SongSeek() broken?

### ISSUE THREAD

**Issue by sezero** [Mindwerks/wildmidi#141]
Using the '<' and '>' keys on the wildmidi player, I can' get SongSeek() to work properly. Tried with kyra1a.xmi and also with the xmi2mid-cenverted version of it. (Since the github idiocy don't support it, I am sending those files to personal emails of @chrisisonwildcode and @psi29a .)
.

**Comment by chrisisonwildcode** [comment]
I'll have a look at this tonight. I get the feeling a fix elsewhere might have broken this.

**Comment by sezero** [comment]
PING ?

**Comment by chrisisonwildcode** [comment]
Working on this later today, might be a good time to point me to anything else I need to look at :)

**Comment by chrisisonwildcode** [comment]
@sezero can you please confirm the issue you are seeing is that at the last song in kyra1a.xmi pressing < does not go back a song. However pressing < repeatedly in quick succession eventually allows it to go back. Pressing < or > before the last song appears to work fine.

That is the issue I am seeing, and tracing at the moment.

**Comment by chrisisonwildcode** [comment]
Found the culprit. I very small track right before the last track.

Output from detest ...
`
New FORM
FORM length: 58
XMID Data
TIMB length: 4
Patch:1, Bank:0
Patch:35, Bank:0

EVENT length: 34
Time 09:55.40782    Meta Event: SMPTE Offset: 60 00 00 00   
Meta data (5 bytes): 
0x60 ` 0x00 0x00 0x00  0x00 
Time 09:55.40782    Controller  Unknown(110): chan(10) set(127)
Time 09:55.40782    Meta Event: Time Signature: 04 02 18 08
Meta data (4 bytes): 
0x04 0x02 0x18 0x08 
Time 09:55.40782    Meta Event: Tempo
Meta data (3 bytes): 
0x07 0xa1 ? 0x20 
Time 09:55.40782    Set Patch: chan(10) patch(35)
Time 09:55.40782    Note On: chan(10) note(45) vel(75)
Note Length: 0.750000 secs
Time 09:56.15782    Meta Event: End Of Track
`

The player (and SongSeek) are working as intended, since the track is less than 1sec it is not the players fault ... we could (for another release) add in a SongNumber function and info but it too would have the same issue with kyra1a.xmi if the info was used to go back a song at the last track in this file.

**Comment by sezero** [comment]
Can we not have a workaround for this?

**Comment by psi29a** [comment]
Does this very small track produce something? I see a set patch and a note on... like a sound effect?

**Comment by chrisisonwildcode** [comment]
A work around in the player could be a key that does 2 reverse songseek's at once.

I could either modify songseek or add a cuesong to allow selection of a specific track number that would probably help developers using the lib, but wouldn't help our player which is a basic way to show how the lib can be used.

As for what it's playing, it's a single note on a bass instrument. I'm guessing used for some effect within the game the file is from.

**Comment by psi29a** [comment]
@sezero is this necessary? If the file has multiple tracks, then the player is doing what is asked of it.

Perhaps an indicator of which track is currently playing in a list of tracks as an visual indicator is a better idea?

**Comment by chrisisonwildcode** [comment]
um, are we waiting on me for this one?

**Comment by sezero** [comment]
I don't know what to do with this one

**Comment by sezero** [comment]
`WONTFIX`, I guess...
